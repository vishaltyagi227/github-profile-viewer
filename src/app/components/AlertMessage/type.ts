import { SUCCESS, ERROR, WARNING, INFO } from '@constant/index';

export interface IAlertMessage {
  open: boolean;
  alertType: typeof SUCCESS | typeof ERROR | typeof WARNING | typeof INFO;
  handleClick?: () => void;
  message: string | null;
  type?: string;
}
