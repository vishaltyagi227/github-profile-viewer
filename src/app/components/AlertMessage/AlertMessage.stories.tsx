import React, { useState } from 'react';
import { Story, Meta } from '@storybook/react';
import Button from '@components/Button';
import AlertMessage from '@components/AlertMessage';
import { IAlertMessage } from '@components/AlertMessage/type';

export default {
  title: 'Components/AlertMessage',
  component: AlertMessage,
} as Meta;

const Template: Story<IAlertMessage> = ({ ...args }) => {
  const [open, setOpen] = useState(false);
  return (
    <>
      <Button
        type="button"
        variant="contained"
        disabled={open}
        onClick={() => {
          setOpen((preState) => !preState);
        }}
      >
        Open Alert
      </Button>
      <AlertMessage
        {...{
          ...args,
          open,
          handleClick: () => {
            setOpen((preState) => !preState);
          },
        }}
      />
    </>
  );
};

export const Alert = Template.bind({});
Alert.args = {
  type: 'alert',
  alertType: 'success',
  message: 'Alert Message',
};

export const Snackbar = Template.bind({});
Snackbar.args = {
  type: 'autohide',
  alertType: 'success',
  message: 'Alert Message',
};
