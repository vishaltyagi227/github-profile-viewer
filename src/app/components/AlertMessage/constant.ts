export const AUTO_HIDE = 'autohide';
export const ALERT = 'alert';
export const ALERT_DURATION = 6000;
