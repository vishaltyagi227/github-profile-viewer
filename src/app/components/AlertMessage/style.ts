import { Theme } from '@mui/material/styles';
import { createStyles, makeStyles } from '@mui/styles';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    closeIcon: {
      color: 'inherit',
    },
    alertWrapper: {
      width: '100%',
    },
    alert: {
      marginBottom: theme.spacing(2),
    },
  })
);

export default useStyles;
