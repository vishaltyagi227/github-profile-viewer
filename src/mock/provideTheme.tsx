import React, { ReactElement } from 'react';
import { ThemeProvider } from '@mui/material/styles';
import theme from '@theme/index';

export default (ui: ReactElement): ReactElement => (
  <ThemeProvider theme={theme}>{ui}</ThemeProvider>
);
