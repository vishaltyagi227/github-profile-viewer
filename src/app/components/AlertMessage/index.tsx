import React from 'react';
import { Collapse, Alert, IconButton, Snackbar } from '@mui/material';
import CloseIcon from '@mui/icons-material/Close';

import { SIZE, ORIGIN } from '@constant/index';
import useStyles from './style';
import { IAlertMessage } from './type';
import { ALERT, AUTO_HIDE, ALERT_DURATION } from './constant';

const AlertMessage = ({
  open,
  alertType,
  message,
  handleClick,
  type,
}: IAlertMessage) => {
  const classes = useStyles();

  if (type === AUTO_HIDE) {
    return (
      <>
        <Snackbar
          open={open}
          autoHideDuration={ALERT_DURATION}
          onClose={handleClick}
          anchorOrigin={{ vertical: ORIGIN.bottom, horizontal: ORIGIN.right }}
        >
          <Alert onClose={handleClick} severity={alertType}>
            {message}
          </Alert>
        </Snackbar>
      </>
    );
  }

  return (
    <>
      <Collapse in={open} className={classes.alertWrapper}>
        <Alert
          severity={alertType}
          action={
            handleClick && (
              <IconButton
                className={classes.closeIcon}
                size={SIZE.small}
                onClick={handleClick}
              >
                <CloseIcon fontSize={SIZE.inherit} />
              </IconButton>
            )
          }
          className={classes.alert}
        >
          {message}
        </Alert>
      </Collapse>
    </>
  );
};

AlertMessage.defaultProps = {
  type: ALERT,
};

export default AlertMessage;
