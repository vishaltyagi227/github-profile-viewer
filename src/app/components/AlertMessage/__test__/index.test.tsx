import React from 'react';
import { cleanup, render, RenderResult } from '@testing-library/react';
import renderer from 'react-test-renderer';
import provideTheme from '@mock/provideTheme';
import AlertMessage from '../index';

describe('<AlertMessage />', () => {
  const message = 'Testing Alert Message Component';
  const mockFn = jest.fn();
  let container: RenderResult;
  beforeEach(() => {
    container = render(
      provideTheme(
        <AlertMessage
          open
          alertType="success"
          message={message}
          handleClick={mockFn}
        />
      )
    );
  });

  afterEach(cleanup);

  it('Should render correctly', () => {
    expect(container.getByRole('alert')).toBeInTheDocument();
  });

  it('Should display message as pass in props', () => {
    expect(container.getByRole('alert').textContent).toBe(message);
  });
});

describe('<AlertMessage /> Snackbar', () => {
  const message = 'Testing Snackbar Message Component';
  const mockFn = jest.fn();
  let container: RenderResult;
  beforeEach(() => {
    container = render(
      provideTheme(
        <AlertMessage
          open
          type="autohide"
          alertType="success"
          message={message}
          handleClick={mockFn}
        />
      )
    );
  });

  afterEach(cleanup);

  it('Should render correctly', () => {
    expect(container.getByRole('alert')).toBeInTheDocument();
  });

  it('Should display message as pass in props', () => {
    expect(container.getByRole('alert').textContent).toBe(message);
  });
});

describe('<AlertMessage /> Snapshot', () => {
  const message = 'Testing Snackbar Message Component';
  const mockFn = jest.fn();

  afterEach(cleanup);

  it('Should match with snapshot', () => {
    const tree = renderer
      .create(
        provideTheme(
          <AlertMessage
            open
            alertType="success"
            message={message}
            handleClick={mockFn}
          />
        )
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
