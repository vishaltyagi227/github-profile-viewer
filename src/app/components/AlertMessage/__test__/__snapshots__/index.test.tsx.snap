// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertMessage /> Snapshot Should match with snapshot 1`] = `
<div
  className="MuiCollapse-root MuiCollapse-vertical makeStyles-alertWrapper-14 MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
  style={
    Object {
      "minHeight": "0px",
    }
  }
>
  <div
    className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
  >
    <div
      className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard makeStyles-alert-15 css-13rvizp-MuiPaper-root-MuiAlert-root"
        role="alert"
      >
        <div
          className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="SuccessOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
            />
          </svg>
        </div>
        <div
          className="MuiAlert-message css-acap47-MuiAlert-message"
        >
          Testing Snackbar Message Component
        </div>
        <div
          className="MuiAlert-action css-ki1hdl-MuiAlert-action"
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall makeStyles-closeIcon-13 css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[MockFunction]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
